File: shipment-ms/shipment-lib/src/main/java/com/flashex/shipmentmicroservice/lib/model/Packet.java
package com.flashex.shipmentmicroservice.lib.model;

import com.datastax.driver.core.DataType;
import lombok.*;
import net.minidev.json.annotate.JsonIgnore;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Table("packet")
@UserDefinedType("packet")
public class Packet {

    /** String variables **/

    @PrimaryKey
    @PrimaryKeyColumn(name = "packetId",  ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    @CassandraType(type = DataType.Name.TEXT)
    private String packetId;
    @CassandraType(type = DataType.Name.TEXT)
    private String productId;
    @CassandraType(type = DataType.Name.TEXT)
    private String packetDescription;
    @CassandraType(type = DataType.Name.TEXT)
    private String deliveryDescription;
    @CassandraType(type = DataType.Name.TEXT)
    private String packetType;
    @CassandraType(type = DataType.Name.TEXT)
    private String priority;
//    /** Float variables **/

    @CassandraType(type = DataType.Name.FLOAT)
    private float weight;
    @CassandraType(type = DataType.Name.FLOAT)
    private float length;
    @CassandraType(type = DataType.Name.FLOAT)
    private float breadth;
    @CassandraType(type = DataType.Name.FLOAT)
    private float height;
    @CassandraType(type = DataType.Name.FLOAT)
    private float costOfPacket;


    /** Date type variables**/
    @CassandraType(type = DataType.Name.TIMESTAMP)
    private Date receivedDate;
//
//    //to be updated by Trip Planning microservice
    @CassandraType(type = DataType.Name.TIMESTAMP)
    private Date estimatedDeliveryDate;
//
//    //to be updated by Trip Itinerary microservice
//    private Date actualDeliveryDate;
//
//    /** Enum type variables **/
//

    /** Objects of same package**/
    @CassandraType(type = DataType.Name.UDT, userTypeName = "deliveryAddress")
    private DeliveryAddress deliveryAddress;

    @CassandraType(type = DataType.Name.UDT, userTypeName = "customer")
    private Customer customer;


    @CassandraType(type = DataType.Name.LIST, typeArguments = { DataType.Name.UDT }, userTypeName = "status")
    private List<Status> statusList = new ArrayList<>();


}

File: common-ms/flashex-webapp/flashex-webapp/src/app/trip-management/interfaces/Packet.ts
export interface IPacket {
  packetId: string;
  productId: string;
  packetDescription: string;
  deliveryDescription: string;
  packetType: string;
  priority: string;
  currentStatus: string;
  receivedDate: string;
   statusList: IStatus[];
  weight: number;
  length: number;
  breadth: number;
  height: number;
  costOfPacket: number;
  estimatedDeliveryDate: Date;
  actualDeliveryDate: Date;
  deliveryAddress: {
      addressLine1: string,
      city: string,
      state: string,
      latitude: number,
      longitude: number,
      pincode: number
  };
  customer: {
      firstName: string,
      middleName: string,
      lastName: string,
      emailId: string,
      phoneNumber: number
  };

  // statusList: [{
  //   statusValue: string;
  //   timeStamp: Date;
  // }];
}

export interface IStatus {
statusValue: string;
timeStamp: string;
}


